### PR TITLE
Live update acquistion 

### PR DESCRIPTION
**Pull request by amuskens**
For spectrum, AR, and temporal spectrum

### CODE REVIEW

File: src/odemis/acq/stream/_sync.py [hunk 1]
```diff
@@ -2053,6 +2054,10 @@ def _runAcquisition(self, future):
                 s = model.DataArray(pxsum, se_md)
                 se_data.append(s)
 
+                # Live update the setting stream with the new data
+                self._se_stream._onNewData(None, se_data)
+                self._tc_stream._onNewData(None, tc_data)
```
Comment (pieleric) [inline src/odemis/acq/stream/_sync.py]: I don't think you should update the "se" (=e-beam) data, as for now this stream is not shown, and if you pass it as-is, it'd show a single pixel.
For the `_tc_stream`, I'd suggest to pass the dataflow as well, just to make sure we don't break anything if the function needs it (in the future). You can use `self._tc_stream._dataflow`.

File: src/odemis/acq/test/stream_test.py [hunk 2]
```diff
@@ -44,6 +44,8 @@
 import unittest
 from unittest.case import skip
 import weakref
+from odemis.model._metadata import MD_ACQ_DATE
+import operator
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: Drop this non-used import.

File: src/odemis/acq/test/stream_test.py [hunk 3]
```diff
@@ -44,6 +44,8 @@
 import unittest
 from unittest.case import skip
 import weakref
+from odemis.model._metadata import MD_ACQ_DATE
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: Just write in the code `model.MD_ACQ_DATE` and drop this import.

File: src/odemis/acq/stream/_sync.py [hunk 4]
```diff
@@ -1107,7 +1107,8 @@ def _runAcquisitionEbeam(self, future):
 
                     # integrate/sum images
                     self._integrateImages(integration_count)
-
+                    # Live update the setting stream with the new data
+                    self._sccd._onNewData(self._ccd_df, self._acq_data[self._ccd_idx][-1])
```
Comment (pieleric) [inline src/odemis/acq/stream/_sync.py]: Please add similar code to the `_runAcquisitionScanStage()`. Somewhere just before the `break` (l 1595) should be equivalent. As the two functions are very similar, it should work well too.

File: src/odemis/driver/simstreakcam.py [hunk 5]
```diff
@@ -274,7 +274,7 @@ def _getNewImage(self):
         """
         self._img_counter = (self._img_counter + 1) % len(self._img_list)
         image = self._img_list[self._img_counter]
-
+        image += (numpy.random.rand(*image.shape) * 1000 - 500).astype(numpy.uint16)
```
Comment (pieleric) [inline src/odemis/driver/simstreakcam.py]: Let's not add this extra debug code here. This driver already supports sending multiple images.

File: src/odemis/driver/andorcam2.py [hunk 6]
```diff
@@ -2954,6 +2954,8 @@ def GetMostRecentImage16(self, cbuffer, size):
         ndbuffer[...] = self._data[self.roi[2] - 1:self.roi[3]:self.binning[1],
                                    self.roi[0] - 1:self.roi[1]:self.binning[0]]
 
+        ndbuffer += (numpy.random.rand(res[1], res[0]) * 1000 - 500).astype(numpy.uint16)
```
Comment (pieleric) [inline src/odemis/driver/andorcam2.py]: Please check that all the test cases still pass (andorcam2_test). I don't think any test case expects that images are always the same, but you need to double check.

Also, I'd suggest to tune down a little bit the values to around 200, which is the typical noise level on these cameras. Also make them only positive, to simplify a little bit. Moreover `randint()` seems to do just what you need. So you could write:
`ndbuffer += numpy.random.randint(0, 200, ndbuffer.shape, dtype=ndbuffer.dtype)`

File: src/odemis/acq/test/stream_test.py [hunk 7]
```diff
@@ -2919,6 +3014,57 @@ def _roiToPhys(self, repst):
...
+            time.sleep(3.0)
+            im2 = ars.image.value
+
+            # wait until it's over
+            data = f.result(timeout)
+
+            # Check if the image changed (live update is working)
+            self.assertRaises(AssertionError, numpy.testing.utils.assert_array_equal, im1, im2)
+
     def test_acq_arpol(self):
```
Comment (pieleric) [inline src/odemis/acq/test/stream_test.py]: Please also add a test for the `SEMTemporalMDStream`.

File: src/odemis/driver/andorcam2.py (hunk 6 above)
Comment (amuskens) [reply src/odemis/driver/andorcam2.py]: The Andorcam tests run fine.

File: src/odemis/acq/test/stream_test.py (hunk 7 above)
Comment (amuskens) [reply src/odemis/acq/test/stream_test.py]: Somehow the backend was not starting so I couldn't verify this test yet.

**Comment by amuskens** [comment]
Ok. Added back the change

**Comment by pieleric** [comment]
I think you forgot to push. I don't see any change.

File: src/odemis/driver/andorcam2.py (hunk 6 above)
Comment (amuskens) [reply src/odemis/driver/andorcam2.py]: I force-pushed it now.

**Comment by pieleric** [comment]
Well, now I *think* the code is correct, but it's back to 5 commits! Please squash the commits into just 2, as discussed previously. There should also be no change to the simstreakcam.

**Comment by amuskens** [comment]
Ahhh... noo.. I will see if  can fix it.

**Pull request merged**